### PR TITLE
SQS-377 | Unit test ProcessPool

### PR DESCRIPTION
**Pull request by deividaspetraitis**
This pull request introduces unit tests for Orderbook usecase `ProcessPool` method. As a side effect for more efficient testing  it refactors several program parts, for example `FetchTickUnrealizedCancels` and `FetchTicks` are moved to the `client` side instead of keeping in `orberbook` usecase.

Tests aims to cover errors and one one successful case:

```
❯ go test ./orderbook/usecase -run "TestOrderbookUsecaseTestSuite/TestProcessPool" -v
=== RUN   TestOrderbookUsecaseTestSuite
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/pool_is_nil
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/cosmWasmPoolModel_is_nil
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/pool_is_not_an_orderbook_pool
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/orderbook_pool_has_no_ticks,_nothing_to_process
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_cast_pool_model_to_CosmWasmPool
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_fetch_ticks_for_pool
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_fetch_unrealized_cancels_for_pool
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/tick_ID_mismatch_when_fetching_unrealized_ticks
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/tick_ID_mismatch_when_fetching_tick_states
=== RUN   TestOrderbookUsecaseTestSuite/TestProcessPool/successful_pool_processing
--- PASS: TestOrderbookUsecaseTestSuite (0.03s)
    --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool (0.03s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/pool_is_nil (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/cosmWasmPoolModel_is_nil (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/pool_is_not_an_orderbook_pool (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/orderbook_pool_has_no_ticks,_nothing_to_process (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_cast_pool_model_to_CosmWasmPool (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_fetch_ticks_for_pool (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/failed_to_fetch_unrealized_cancels_for_pool (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/tick_ID_mismatch_when_fetching_unrealized_ticks (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/tick_ID_mismatch_when_fetching_tick_states (0.00s)
        --- PASS: TestOrderbookUsecaseTestSuite/TestProcessPool/successful_pool_processing (0.00s)
PASS
ok      github.com/osmosis-labs/sqs/orderbook/usecase   (cached)
```

### CODE REVIEW

File: orderbook/usecase/orderbook_usecase_test.go
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Observing same structs to keep repeating. Consider defining a default mock routable pool and using a decorator pattern to modify fields that are necessary for the tests:
```go
func withPoolID(mock *mocks.RoutablePool, poolID uin64) *mocks.RoutablePool {
   mock.PoolID = poolID
   return mock
}
```

File: orderbook/usecase/orderbook_usecase_test.go
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Let's also import the actual errors rather than hardcoding strings.

This is so that if we need to change an error message, we only need to do it in one place

**Review by p0mvn — COMMENTED** [review]
Overall LGTM!

My only recommendation is to reduce code duplication and hard coding before merge.

File: orderbook/usecase/orderbook_usecase_test.go
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: I have moved forward with your suggested approach an implemented wrappers, it looks much cleaner now. :+1:

File: orderbook/usecase/orderbook_usecase_test.go
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: This is actually the identical situation as described in previous PR [comment](https://github.com/osmosis-labs/sqs/pull/492#discussion_r1738497416). If you would like still to proceed with this idea can we descope it from this task?

Comment (p0mvn) [reply orderbook/usecase/orderbook_usecase_test.go]: Per the comment in [another PR](https://github.com/osmosis-labs/sqs/pull/492#discussion_r1740168398), I would like to suggest considering completing it in this PR so that we avoid accumulating tech debt.

IMO, this should be under 20 minutes to change/refactor but will yield the benefit of not having to switch context to come back to in the future and/or accumulate tech debt that is already exceeding the comfortable levels.

Will leave the final call to you though!

**Review by p0mvn — APPROVED** [review]
Pre-approving pending the decision on handling errors

**Comment by deividaspetraitis** [comment]
As per suggestion implemented error types instead of [hardcoding](https://github.com/osmosis-labs/sqs/pull/493/files#diff-a585202576c1e374948b8b760a1fe96534e5a35a29ccb3e846e5a7548c714f0aR136) strings. Thanks @p0mvn for your time and suggestions! :pray:

**Pull request merged**
